integration_tests/integration/court-register-list.spec.ts

```diff
@@ -0,0 +1,104 @@
+    cy.task('stubPageOfCourts', {
+      content: [sheffieldCrownCourt, { ...sheffieldMagistratesCourt, active: true }, sheffieldYouthCourt],
+      last: false,
+      totalPages: 2,
+      totalElements: 4,
+      number: 0,
+      size: 3,
+      first: true,
+      numberOfElements: 3,
+      empty: false,
+    })
+    cy.task('stubCourt', sheffieldCrownCourt)
+    cy.task('stubCourt', sheffieldMagistratesCourt)
+    cy.login()
+  })
+
+  it('Can navigate to court registers', () => {
+    IndexPage.verifyOnPage().courtRegisterLink().should('contain.text', 'Court register').click()
+    AllCourtsPage.verifyOnPage()
+  })
+
+  it('Will display all courts', () => {
+    IndexPage.verifyOnPage().courtRegisterLink().click()
+    const courtRegisterPage = AllCourtsPage.verifyOnPage()
+
+    {
+      const { code, name, type, status } = courtRegisterPage.courts(0)
+      code().contains('SHFCC')
+      name().contains('Sheffield Crown Court')
+      type().contains('Crown')
+      status().contains('Active')
+    }
+    {
+      const { code, name, type, status } = courtRegisterPage.courts(1)
+      code().contains('SHFMC')
+      name().contains('Sheffield Magistrates Court')
+      type().contains('Magistrates')
+      status().contains('Closed')
+    }
+  })
+
+  it('Will display a page of courts', () => {
+    // IndexPage.verifyOnPage().courtRegisterLink().click()  -  TODO Will need this when plumbing in the paged court list and removing the all courts list
+    IndexPage.verifyOnPage()
+    cy.visit('/court-register/paged') // TODO and this will need removing
+    const courtRegisterPagedPage = AllCourtsPagedPage.verifyOnPage()
+
+    {
+      const { code, name, type, status } = courtRegisterPagedPage.courts(0)
+      code().contains('SHFCC')
+      name().contains('Sheffield Crown Court')
+      type().contains('Crown')
+      status().contains('Active')
+    }
+    {
+      const { code, name, type, status } = courtRegisterPagedPage.courts(1)
+      code().contains('SHFMC')
+      name().contains('Sheffield Magistrates Court')
+      type().contains('Magistrates')
+      status().contains('Active')
+    }
+    {
+      const { code, name, type, status } = courtRegisterPagedPage.courts(2)
+      code().contains('SHFYC')
+      name().contains('Sheffield Youth Court')
+      type().contains('Youth')
+      status().contains('Closed')
+    }
+  })
+  it('Will display pagination controls', () => {
+    // IndexPage.verifyOnPage().courtRegisterLink().click()  -  TODO Will need this when plumbing in the paged court list and removign the all courts list
+    IndexPage.verifyOnPage()
+    cy.visit('/court-register/paged') // TODO and this will need removing
+    const courtRegisterPagedPage = AllCourtsPagedPage.verifyOnPage()
+
+    const pageLinks = courtRegisterPagedPage.pageLinks()
+    pageLinks.then(items => {
+      // TODO There must be a better way than this?
+      expect(items[0].href).to.equal(undefined)
```

Comment:
should be able to compare all at once e.g.
```
expect(items[0]).to.equal({href: undefined, text: '1', selected: true})
```
or maybe even all items in one go...

Reply:
yes, should be a way